macros/seed_existance_check.sql

```diff
@@ -0,0 +1,38 @@
+{#
+Copyright (c) 2023-present Snowplow Analytics Ltd. All rights reserved.
+This program is licensed to you under the Snowplow Personal and Academic License Version 1.0,
+and you may not use this file except in compliance with the Snowplow Personal and Academic License Version 1.0.
+You may obtain a copy of the Snowplow Personal and Academic License Version 1.0 at https://docs.snowplow.io/personal-and-academic-license-1.0/
+#}
+
+{% macro seed_existance_check() %}
+    {% if execute %}
+        {# Ensure that this check is only performed during 'run' or 'build' commands #}
+        {%- if flags.WHICH in ('run') -%}
+            {% for node in graph.nodes.values() | selectattr("resource_type", "equalto", "seed") %}
+
+                {# Convert schema and table names to uppercase to prevent case sensitivity issues #}
+                {% set schema = node.schema | upper %}
+                {% set table = node.alias | upper %}
+                
+                {# Construct a SQL query to check if the seed table exists in the information schema #}
+                {% set check_table_exists_query = """SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = '" ~ schema ~ "' AND table_name = '" ~ table ~ "'" %}
+                
+                {% set results = run_query(check_table_exists_query) %}
+                {% set table_exists = results.columns[0][0] if results and results.columns[0] else 0 %}
+                
+                {# Log an error and fail the run if the table does not exist #}
+                {% if table_exists | int == 0 %}
+                    {% do log(node.unique_id ~ " does not exist.", info=true) %}
+
+                    {# This is a bad way to handle this, but it's the only way to fail a dbt macro
+                    running in the on-run-start hook while in execution mode. We do this until the update #}
+                    
+                    {{ 0/0 }}
+
+                {% endif %}
+            {% endfor %}
+        {%- endif %}
```

Comment:
I understand that currently there is a not a way to raise a full stopping error from a on-run-start hook, but I personally really hate this approach. It leads to a really long error trace that in most consoles will actually be long enough to push the log above off screen - and given we're trying to make it easier for less dbt capable people, they may not end up looking back up.

I think here we have 2 realistic options:

1. Keep it as an on-run-start hook, raise an exception with a 3 line error (see example here https://github.com/snowplow/dbt-snowplow-utils/blob/7ed19bcbb5df41f0647f5a832b9fee9653a52d9d/macros/utils/return_limits_from_model.sql#L35) and just accept it won't stop running until people are on 1.8
2. Can we test what happens if we run this as a pre-hook on a _model_. We know the first model in our DAG is the incremental manifest table, if we add this as a pre-hook to that would raising an exception correctly stop the run?

Reply:
You can see it in the DAG here https://snowplow.github.io/dbt-snowplow-unified/#!/overview/snowplow_unified?g_v=1

Give it a shot and see if it works, we can always move it out into an on-start-hook later once dbt has supported that approach for longer.